float32 initial_energy
float32 decay
float32 cell_size
float32[] x_coordinates
float32[] y_coordinates
float32[] energy
